src/Compilers/CSharp/Portable/Lowering/BoundTreeToDifferentEnclosingContextRewriter.cs

```diff
@@ -57,7 +59,22 @@ protected virtual bool TryRewriteLocal(LocalSymbol local, [NotNullWhen(true)] ou
             var newType = VisitType(local.Type);
             if (TypeSymbol.Equals(newType, local.Type, TypeCompareKind.ConsiderEverything2))
```

Comment:
>if (TypeSymbol.Equals(newType, local.Type, TypeCompareKind.ConsiderEverything2)) [](http://example.com/codeflow?start=12&length=80)

It looks like the code could be simplified as follows:
```
            if (TypeSymbol.Equals(newType, local.Type, TypeCompareKind.ConsiderEverything2) &&
                (!EnforceAccurateContainerForLocals || local.ContainingSymbol == CurrentMethod))
            {
                newLocal = local;
            }
            else
            {
                newLocal = new TypeSubstitutedLocalSymbol(local, TypeWithAnnotations.Create(newType), CurrentMethod);
                localMap.Add(local, newLocal);
            }
```

Reply:
Like jcouv said - I don't think `local.ContainingSymbol == CurrentMethod` is ever true. The whole class is about changing the enclosing context...

Reply:
> Like jcouv said - I don't think ```local.ContainingSymbol == CurrentMethod``` is ever true. The whole class is about changing the enclosing context...

Where did he say that? I only saw him asking you to explain the logic.

Second. This is a base class and you cannot really predict how it is going to be used in the future. Please explain why would we want to create new symbols when nothin about them is changing?